packages/block-library/src/post-excerpt/index.php

```diff
@@ -18,12 +18,18 @@ function render_block_core_post_excerpt( $attributes, $content, $block ) {
 		return '';
 	}
 
-	$excerpt = get_the_excerpt();
-
-	if ( empty( $excerpt ) ) {
-		return '';
+	/*
+	* The purpose of the excerpt length setting is to limit the length of both
+	* automatically generated and user-created excerpts.
+	* Because the excerpt_length filter only applies to auto generated excerpts,
+	* wp_trim_words is used instead.
+	*/
+	$excerpt_length = $attributes['excerptLength'];
+	if ( isset( $excerpt_length ) ) {
+		$excerpt = wp_trim_words( get_the_excerpt(), $excerpt_length );
+	} else {
+		$excerpt = get_the_excerpt();
```

Comment:
Why not inject a filter with the dynamic excerpt length and just call `get_the_excerpt`, which is [already filtered](https://github.com/WordPress/wordpress-develop/blob/b55f01db1dd0c8de654e4175121337fe25f24dca/src/wp-includes/default-filters.php#L200) by `wp_trim_excerpt`?

Reply:
I don't believe I tried that during the development, so there is no obvious reason for not trying that as a possible improvement in a separate PR.

Reply:
The addition of the call to "wp_trim_words" will remove any HTML that users have put into a custom excerpt by over-riding get_the_excerpt.  Was this deliberate, or an unfortunate by-product?

Reply:
Woraround to bypass the `wp_trim_words` function:

```
function fzs_filter_metadata_registration( $metadata ) {
    if ($metadata['name'] === 'core/post-excerpt') {
	unset($metadata['attributes']['excerptLength']);
    }
    return $metadata;
};
add_filter( 'block_type_metadata', 'fzs_filter_metadata_registration' );
```

Reply:
Thank you - that's a really helpful workaround for now

Reply:
This was also reported [in the forums](https://wordpress.org/support/topic/excerpt-stripping-out-html/) and I've confirmed it was introduced in WordPress 6.3. Created an issue at https://github.com/WordPress/gutenberg/issues/66195